impl/client/src/main/javascript/web/util/URLEncoder.js

```diff
@@ -70,22 +78,24 @@ define("common-ui/util/URLEncoder", [], function() {
       args = [args];
     }
 
-    // Detect the presence of the "?" to determine
-    // when the special double-slash encoding should end.
-    var pathPart = str.split("?")[0];
-    var pathBounds = (pathPart.match(/\{[\d]+\}/g) || []).length;
+    if (doubleEncodeSlashes) {
+      // Detect the presence of the "?" to determine
+      // when the special double-slash encoding should end.
+      var pathPart = str.split("?")[0];
+      var pathBounds = (pathPart.match(/\{[\d]+\}/g) || []).length;
 
-    args = (args || []).map(function(item, pos) {
+      args = (args || []).map(function (item, pos) {
 
-      var encodedStr = encodeURIComponent(String(item));
+        var encodedStr = encodeURIComponent(String(item));
 
-      // Double-encode / and \ to work around Tomcat issue.
-      if(pos < pathBounds) {
-        encodedStr = encodedStr.replace(/%5C/g, "%255C").replace(/%2F/g, "%252F");
-      }
+        // Double-encode / and \ to work around Tomcat issue.
+        if (pos < pathBounds) {
+          encodedStr = encodedStr.replace(/%5C/g, "%255C").replace(/%2F/g, "%252F");
+        }
```

Comment:
The double encoding `doubleEncodeSlashes` parameter should only affect the execution of this block, and not of the entire loop, which includes the necessary `encodeURIComponent` call.